src/Adapter/Image/ImageRetriever.php

```diff
@@ -203,7 +203,7 @@ public function getImage($object, $id_image)
             }
 
             $url = $this->link->$getImageURL(
-                isset($object->link_rewrite) ? $object->link_rewrite : $object->name,
+                !empty($object->link_rewrite) ? $object->link_rewrite : (!empty($object->name) ? $object->name : $id_image),
```

Comment:
Nothing really wrong here, just a bit overcomplicated to chain ternary operators this way.
Functional cases are usually better suited in a good old `if` structure.

Reply:
```suggestion
                $object->link_rewrite ?: $object->name ?: $id_image,
```

Reply:
@Progi1984 Unfortunately chained ternary operators can't handle undefined properties and might crash. Did you mean this ?
```suggestion
                $object->link_rewrite ?? $object->name ?? $id_image
```
This last suggestion can't handle tricky cases either, like empty strings.
IMHO there is no way to replace empty tests with chained operators, because `empty()` can handle both undefined and falsey values.
That's why I suggested if/else structure to handle each functional case only to improve readibility.

Reply:
Oh yeah ! I don't think.

@Hlavtox Use @saulaski solution ;)

Reply:
@Hlavtox We mean the code should be easier to read, no fancy syntax required :)
Something like this:
```suggestion
            if (!empty($object->link_rewrite)) {
                $imageName = $object->link_rewrite;
            } elseif (!empty($object->name) {
                $imageName = $object->name;
            } else {
                $imageName = $id_image;
            }
            $url = $this->link->$getImageURL(
                $imageName,
```